cfme/common/provider.py

```diff
@@ -515,13 +545,14 @@ def process_credential_yaml_key(cls, cred_yaml_key, cred_type=None):
 
     @staticmethod
     def clear_provider_by_type(prov_class, validate=True):
+        # TODO I don't like this here + the name & params combo
```

Comment:
I prefer the last one, honestly. In addition, `utils.providers.list_providers` can turn into a `find` classmethod, which will not need categories nor types, only filters.... I guess.... hmmmm.